frontend/packages/console-dynamic-plugin-sdk/src/utils/fetch/console-fetch.ts

```diff
@@ -27,22 +27,29 @@ export const consoleFetch: ConsoleFetch = async (url, options = {}, timeout = 60
   return Promise.race([fetchPromise, timeoutPromise]);
 };
 
+const parseData = async (response) => {
+  const text = await response.text();
+  const isPlainText = response.headers.get('Content-Type') === 'text/plain';
+  if (!text) {
+    return isPlainText ? '' : {};
+  }
+  return isPlainText || !response.ok ? text : JSON.parse(text);
+};
+
 const consoleFetchCommon = async (
   url: string,
   method: string = 'GET',
   options: RequestInit = {},
   timeout?: number,
-) => {
+  isEntireResponse?: boolean,
+): Promise<Response> => {
   const headers = getConsoleRequestHeaders();
+  const defaultIsEntireResponse = typeof isEntireResponse !== 'boolean' ? false : isEntireResponse;
```

Comment:
@spadgett @TheRealJon 
Follow on to my justification for adding this logic to enforce  type check in `consoleFetchCommon` to prevent breaking change by consumers.  The [k8sCreate](https://github.com/openshift/console/blob/master/frontend/public/components/edit-yaml.jsx#L334) method instance in edit-yaml.jsx takes four args. Seems `name` and `namespace` are unused. These arguments in `k8sCreate` would cause the `consoleFetchCommon` method to return the entire response instead of the response body which would break the Create Resource YAML functionality. I have opened a bug to investigate and fix the issue in the console. But not sure how we could handle this situation if a plugin happens to be doing the same thing we are doing here if we don’t enforce the type to be boolean.

Reply:
We should probably address the root cause in `edit-yaml.jsx` rather than code a workaround for the new behavior.